WooCommerce/Classes/ViewRelated/Products/Edit Product/Inventory Settings/ProductInventorySettingsViewController.swift

```diff
@@ -336,11 +338,14 @@ private extension ProductInventorySettingsViewController {
 //
 private extension ProductInventorySettingsViewController {
     @objc func scanSKUButtonTapped() {
-        let scannerViewController = ProductSKUInputScannerViewController(onBarcodeScanned: { [weak self] barcode in
+        guard let navigationController = navigationController else {
+            return
+        }
+        let coordinator = ProductSKUBarcodeScannerCoordinator(sourceNavigationController: navigationController) { [weak self] barcode in
             self?.onSKUBarcodeScanned(barcode: barcode)
-            self?.navigationController?.popViewController(animated: true)
-        })
-        show(scannerViewController, sender: self)
+        }
+        skuBarcodeScannerCoordinator = coordinator
+        coordinator.start()
```

Comment:
Suggestion: maybe we should dismiss the keyboard here before starting the scanner? I think it'd be nicer than this:
![IMG_0062](https://user-images.githubusercontent.com/5533851/146296082-05d415ea-f195-4170-9513-679fff8fc721.PNG)

Reply:
Good suggestion! Added in https://github.com/woocommerce/woocommerce-ios/pull/5679/commits/3496f277d439d5bcdf52170e7702a1ab4eeb4108